test/src/test/java/org/jvnet/hudson/main/JenkinsRuleTimeoutTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jvnet.hudson.main;
+
+import hudson.model.listeners.ItemListener;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.Rule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+
+public class JenkinsRuleTimeoutTest {
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+    {
+        r.timeout = 10; // let us not wait three minutes!
+    }
+
+    @Test
+    public void hangInterruptiblyInTest() throws Exception {
+        try {
+            hangInterruptibly();
+        } catch (InterruptedException x) {
+            System.err.println("Interrupted, good.");
+        }
+    }
+
+    @Ignore("TODO ought to use the idiom from org.junit.internal.runners.statements.FailOnTimeout: " +
+            "run test in its own thread so we can fail even if it does not respond to interruptions")
+    @Test
+    public void hangUninterruptiblyInTest() throws Exception {
+        hangUninterruptibly();
+    }
+
+    @Test
+    public void hangInterruptiblyInShutdown() throws Exception {
+        System.err.println("Test itself passed…");
+    }
+    @TestExtension("hangInterruptiblyInShutdown")
```

Comment:
Intentional, since the extension applies only to the method directly above, and Java offers no way of making such scopes explicit.